pkg/agent/baker.go

```diff
@@ -237,6 +237,26 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
 			}
 			return kc.GetOrderedKubeletConfigStringForPowershell()
 		},
+		"ShouldConfigSysctl": func() bool {
+			return profile.CustomLinuxOSConfig != nil && profile.CustomLinuxOSConfig.Sysctls != nil
+		},
+		"ShouldConfigTransparentHugePage": func() bool {
+			return profile.CustomLinuxOSConfig != nil && (profile.CustomLinuxOSConfig.TransparentHugePageEnabled != "" || profile.CustomLinuxOSConfig.TransparentHugePageDefrag != "")
+		},
+		"GetTransparentHugePageEnabled": func() string {
+			return profile.CustomLinuxOSConfig.TransparentHugePageEnabled
+		},
+		"GetTransparentHugePageDefrag": func() string {
+			return profile.CustomLinuxOSConfig.TransparentHugePageDefrag
+		},
+		"ShouldConfigSwapFile": func() bool {
+			// only configure swap file when FailSwapOn is true and SwapFileSizeMB is valid
+			return profile.CustomKubeletConfig != nil && profile.CustomKubeletConfig.FailSwapOn != nil && *profile.CustomKubeletConfig.FailSwapOn &&
+				profile.CustomLinuxOSConfig != nil && profile.CustomLinuxOSConfig.SwapFileSizeMB != nil && *profile.CustomLinuxOSConfig.SwapFileSizeMB > 0
+		},
+		"GetSwapFileSizeMB": func() int32 {
+			return *profile.CustomLinuxOSConfig.SwapFileSizeMB
```

Comment:
CustomLinuxOSConfig and SwapFileSizeMB are pointer types right?

Reply:
yes this func is intended to get the SwapFileSizeMB value